clippy_lints/src/booleans.rs

```diff
@@ -458,12 +456,9 @@ impl<'a, 'tcx> Visitor<'tcx> for NonminimalBoolVisitor<'a, 'tcx> {
 }
 
 
-fn implements_ord<'a, 'tcx>(cx: &'a LateContext<'a, 'tcx>, expr: &Expr) -> Option<bool> {
+fn implements_ord<'a, 'tcx>(cx: &'a LateContext<'a, 'tcx>, expr: &Expr) -> bool {
     let ty = cx.tables.expr_ty(expr);
-
-    return if let Some(id) = get_trait_def_id(cx, &paths::ORD) {
-        Some(implements_trait(cx, ty, id, &[]))
-    } else {
-        None
-    };
+    get_trait_def_id(cx, &paths::ORD)
+        .map(|id| implements_trait(cx, ty, id, &[]))
```

Comment:
heads up: miri dogfood will tell you to turn this into `map_or(false, |id|....`

Reply:
Is this a hint miri directly outputs if you run it on this repo? I spent a few minutes to check if I can get it running but even if I use the intermediate xargo step it seems like I miss something as miri still fails with `no mir for std::env::current_dir` for `cargo +nightly miri test`.